internal/services/network/application_gateway_resource.go

```diff
@@ -4460,3 +4582,165 @@ func applicationGatewayCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceD
+		if v, ok := m["pick_host_name_from_backend_address"]; ok {
+			buf.WriteString(fmt.Sprintf("%t", v.(bool)))
+		}
+		if v, ok := m["request_timeout"]; ok {
+			buf.WriteString(fmt.Sprintf("%d", v.(int)))
+		}
+		if authCert, ok := m["authentication_certificate"].([]interface{}); ok {
+			for _, ac := range authCert {
+				config := ac.(map[string]interface{})
+				buf.WriteString(config["name"].(string))
+			}
+		}
+		if connectionDraining, ok := m["connection_draining"].([]interface{}); ok {
+			for _, ac := range connectionDraining {
+				config := ac.(map[string]interface{})
+				buf.WriteString(fmt.Sprintf("%t", config["enabled"].(bool)))
+				buf.WriteString(fmt.Sprintf("%d", config["drain_timeout_sec"].(int)))
+
+			}
+		}
+		if trustedRootCertificateNames, ok := m["trusted_root_certificate_names"]; ok {
+			buf.WriteString(fmt.Sprintf("%s", trustedRootCertificateNames.([]interface{})))
+		}
+	}
+
+	return pluginsdk.HashString(buf.String())
+}
+
+func applicationGatewaySSLCertificate(v interface{}) int {
+	var buf bytes.Buffer
+	if m, ok := v.(map[string]interface{}); ok {
+		buf.WriteString(m["name"].(string))
+
+		if v, ok := m["data"]; ok {
+			buf.WriteString(v.(string))
+		}
+		if v, ok := m["password"]; ok {
+			buf.WriteString(v.(string))
+		}
+		if v, ok := m["key_vault_secret_id"]; ok {
+			buf.WriteString(v.(string))
+		}
+	}
+
+	return pluginsdk.HashString(buf.String())
+}
+
+func applicationGatewayBackendAddressPool(v interface{}) int {
+	var buf bytes.Buffer
+	if m, ok := v.(map[string]interface{}); ok {
+		buf.WriteString(m["name"].(string))
+
+		if fqdns, ok := m["fqdns"]; ok {
+			if features.ThreePointOh() {
+				buf.WriteString(fmt.Sprintf("%s", fqdns.(*pluginsdk.Set).List()))
+			} else {
+				buf.WriteString(fmt.Sprintf("%s", fqdns.([]interface{})))
+			}
+		}
+		if ips, ok := m["ip_addresses"]; ok {
+			if features.ThreePointOh() {
+				buf.WriteString(fmt.Sprintf("%s", ips.(*pluginsdk.Set).List()))
+			} else {
+				buf.WriteString(fmt.Sprintf("%s", ips.([]interface{})))
+			}
+		}
+	}
+
+	return pluginsdk.HashString(buf.String())
+}
+
+func applicationGatewayProbeHash(v interface{}) int {
+	var buf bytes.Buffer
+	if m, ok := v.(map[string]interface{}); ok {
+		buf.WriteString(m["name"].(string))
+		buf.WriteString(m["protocol"].(string))
+		buf.WriteString(m["path"].(string))
+		buf.WriteString(fmt.Sprintf("%d", m["interval"].(int)))
+		buf.WriteString(fmt.Sprintf("%d", m["timeout"].(int)))
+		buf.WriteString(fmt.Sprintf("%d", m["unhealthy_threshold"].(int)))
+
+		if v, ok := m["host"]; ok {
+			buf.WriteString(v.(string))
+		}
+		if v, ok := m["port"]; ok {
+			buf.WriteString(fmt.Sprintf("%d", v.(int)))
+		}
+		if v, ok := m["pick_host_name_from_backend_http_settings"]; ok {
+			buf.WriteString(fmt.Sprintf("%t", v.(bool)))
+		}
+		if v, ok := m["minimum_servers"]; ok {
+			buf.WriteString(fmt.Sprintf("%d", v.(int)))
+		}
+		// todo do we have to ignore the list attribute `match` because it is optional/computed?
```

Comment:
that would mean that we can't track changes to it - if it's Optional & Computed then it can't be in a Set - can we remove Computed from it?

Reply:
We can't. If you submit an empty probe, you get the following back

```
"match": {
  "body": "",
  "statusCodes": [
  "200-399"]
},
```